src/drivers/mct/main/component_mod.F90

```diff
@@ -63,6 +63,8 @@ module component_mod
   integer  :: mpicom_GLOID, mpicom_CPLID           ! GLOID, CPLID mpi communicator
   integer  :: nthreads_GLOID, nthreads_CPLID
   logical  :: drv_threading
+  integer (kind=MPI_ADDRESS_KIND) :: max_mpi_tag
```

Comment:
Are you sure that this is the correct type for this variable - from the mpich documentation:
The attribute_val in Fortran is a pointer to a Fortran integer, not a pointer to a void *.
Which I think implies that what is in max_mpi_tag is the address of the value and not the value itself?

Reply:
No I'm not a 100% sure. From my understanding though, this is returning the actual value not an address of the value. 

[This mpi-forum page](https://www.mpi-forum.org/docs/mpi-2.2/mpi22-report/node369.htm) explains it and provides some examples, but here is the relevant text: 

>MPI supports two types of attributes: address-valued (pointer) attributes, and integer valued attributes. C and C++ attribute functions put and get address valued attributes. Fortran attribute functions put and get integer valued attributes. When an integer valued attribute is accessed from C or C++, then MPI_xxx_get_attr will return the address of (a pointer to) the integer valued attribute, which is a pointer to MPI_Aint if the attribute was stored with Fortran MPI_xxx_SET_ATTR, and a pointer to int if it was stored with the deprecated Fortran MPI_ATTR_PUT. When an address valued attribute is accessed from Fortran, then MPI_xxx_GET_ATTR will convert the address into an integer and return the result of this conversion. This conversion is lossless if new style attribute functions are used, and an integer of kind MPI_ADDRESS_KIND is returned. The conversion may cause truncation if deprecated attribute functions are used. In C, the deprecated routines MPI_Attr_put and MPI_Attr_get behave identical to MPI_Comm_set_attr and MPI_Comm_get_attr.

And that same behavior is implied here as well:

https://www.open-mpi.org/doc/v3.0/man3/MPI_Comm_set_attr.3.php

Reply:
thanks